src/extensions/surveys.tsx

```diff
@@ -444,6 +445,51 @@ export function usePopupVisibility(
         }
     }, [])
 
+    // Add URL change listener to hide survey when URL no longer matches
+    useEffect(() => {
+        if (isPreviewMode || !survey.conditions?.url || !posthog) {
+            return
+        }
+
+        const checkUrlMatch = () => {
+            const urlMatchType = survey.conditions?.urlMatchType ?? 'icontains'
+            const url = survey.conditions?.url
+            if (!url) {
+                return
+            }
+            const urlCheck = surveyUrlValidationMap[urlMatchType](url)
+            if (!urlCheck) {
+                setIsPopupVisible(false)
+                removeSurveyFromFocus(survey.id)
+            }
+        }
+
+        // Listen for browser back/forward and hash changes (for hash-based routing)
+        window.addEventListener('popstate', checkUrlMatch)
+        window.addEventListener('hashchange', checkUrlMatch)
```

Comment:
> The hashchange event is fired when the fragment identifier of the URL has changed (the part of the URL beginning with and following the # symbol).

add a comment about the hashchange, it wasn't obvious to me

Reply:
added!